lib/util/cidr.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Util
+
+  # provides helper methods for interacting with CIDR network addresses 
+  module CIDR
```

Comment:
Something about this wasn't sitting right with me -- utility classes usually don't -- and I was kind of mulling it over in the back of my mind and trying to figure out why, and then it popped out at me... you have an object here begging to be born.  It's an upgraded version of ruby's `IPAddr` class -- with these additional methods.  So rather than working with plain Strings and sending them to this utility methods to do what you want, make it so you can `cidr = Cidr.new(ip_addr)` and now your client code can `cidr.to_s` (or `cidr.formatted_string`) and `cidr.mask`.

Reply:
That's fair. I'll dig into that some more. Thanks!